src/components/carousel/index.js

```diff
@@ -151,7 +152,10 @@ export default class Carousel extends BaseComponent {
 
   onOrientationChanged = () => {
     const {pageWidth, loop} = this.getThemeProps();
-    if (!pageWidth || loop) {
+    if (pageWidth) {
+      this.orientationChange = true;
+      this.setState({pageWidth});
+    } else if (!pageWidth || loop) {
       this.orientationChange = true;
       // HACK: setting to containerWidth for Android's call when view disappear
       this.setState({pageWidth: this.state.containerWidth || Constants.screenWidth});
```

Comment:
something with the condition here seem off.
first of all `this.orientationChange = true;` always being set to true, so not sure why it's inside the condition twice. 
second.. the `... || loop` seem redundant now. 
if `pageWidth` is defined it will go to the first condition, otherwise it will always enter the second condition without even checking the `loop` value.. 

maybe we're missing something here?

Reply:
I'm not sure what this fixed, but it seems it's not needed anymore - removed